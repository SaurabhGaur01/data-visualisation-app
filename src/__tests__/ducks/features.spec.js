import {
    setFeatures,
    clearFeatures,
    featuresReducer,
} from '../../ducks/features';

describe('features Duck Tests', () => {
    const mockObject = "mockValue";

    const unknownAction = {
        type: 'UNKNOWN',
    }

    describe('Action Generator', () => {
        it('setFeatures should match the snapshot' , () => {
            expect(setFeatures(mockObject)).toMatchSnapshot();
        });
        it('clearFeatures should match the snapshot' , () => {
            expect(clearFeatures(mockObject)).toMatchSnapshot();
        });
    });

    describe('Reducer', () => {
        it('should return the initial state if called with undefined' , () => {
            expect(featuresReducer()).toEqual([]);
        });
        it('should return the passed state if called with no action' , () => {
            expect(featuresReducer(mockObject)).toEqual(mockObject);
        });
        it('should return the passed state if called with an unknown action' , () => {
            expect(featuresReducer(mockObject, unknownAction)).toEqual(mockObject);
        });
        it('should return the value which is passed' , () => {
            expect(featuresReducer('', setFeatures(mockObject))).toEqual(mockObject);
        });
        it('should reset the value to []' , () => {
            expect(featuresReducer('', clearFeatures())).toEqual([]);
        });
    });
});