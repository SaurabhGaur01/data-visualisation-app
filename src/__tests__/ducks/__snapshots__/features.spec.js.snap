// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`features Duck Tests Action Generator clearFeatures should match the snapshot 1`] = `
Object {
  "type": "modules/data-visualisation-app/CLEAR_FEATUREs",
}
`;

exports[`features Duck Tests Action Generator setFeatures should match the snapshot 1`] = `
Object {
  "data": "mockValue",
  "type": "modules/data-visualisation-app/SET_FEATURES",
}
`;
